### PR TITLE
enable 6 modeling cases on XPU

### PR DESCRIPTION
**Pull request by yao-matrix**
**4 PASSED**
PASSED tests/models/mpt/test_modeling_mpt.py::MptIntegrationTests::test_model_logits
PASSED tests/models/nemotron/test_modeling_nemotron.py::NemotronIntegrationTest::test_nemotron_8b_generation_eager
PASSED tests/models/bamba/test_modeling_bamba.py::BambaModelIntegrationTest::test_simple_batched_generate_with_padding
PASSED tests/models/bamba/test_modeling_bamba.py::BambaModelIntegrationTest::test_simple_generate

**2 FAILED**
FAILED tests/models/gemma/test_modeling_gemma.py::GemmaIntegrationTest::test_compile_static_cache, will fix in separate PR
FAILED tests/models/nemotron/test_modeling_nemotron.py::NemotronIntegrationTest::test_nemotron_8b_generation_sdpa, will fix in separate PR

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Hi @yao-matrix, we probably don't want to add tests that we know are failing! I think it works better if you only add/enable them in the PR where you fix them. cc @ydshieh

File: tests/models/nemotron/test_modeling_nemotron.py [hunk 1]
```diff
@@ -202,9 +203,17 @@ def test_nemotron_8b_generation_sdpa(self):
...
+            {
+                ("xpu", 3): [
+                    "What is the largest planet in solar system?\nAnswer: Jupiter\n\nWhat is the answer: What is the name of the 19",
+                ],
+                ("cuda", 7): [
+                    "What is the largest planet in solar system?\nAnswer: Jupiter\n\nWhat is the answer",
+                ],
+            }
+        )
+        EXPECTED_TEXT = EXPECTED_TEXTS.get_expectation()
```
Comment (ydshieh) [inline tests/models/nemotron/test_modeling_nemotron.py:216]: similar to what I mentioned in another PR, I think this is going to break on ROCM? cc @ivarflakstad

Comment (ivarflakstad) [reply tests/models/nemotron/test_modeling_nemotron.py:216]: Might. Might not. We'll see when we run it hehe

Comment (yao-matrix) [reply tests/models/nemotron/test_modeling_nemotron.py:216]: i'll say this "keep the same behavior on rocm without and with this PR", since the `get_expectation` will fallback to use cuda's if we don't set specific expectations for rocm, which is the same expectation as before. So, if it fail with this PR, it will also fail in before. But we have an advantage with this PR, we can add rocm item easily to fix it since we already set up the `Expectations` based code here

**Comment by yao-matrix** [comment]
> Hi @yao-matrix, we probably don't want to add tests that we know are failing! I think it works better if you only add/enable them in the PR where you fix them. cc @ydshieh

@Rocketknight1 , my current practice is:
1. If the cases failed and the bug is in Hugging Face libraries or testing adaptation, I will enable them and fix them.
2. If the cases failed and we triaged the bug is in Hugging Face dependent libraries(for example: PyTorch XPU op issue), I will discuss with our PyTorch team and ask them to fix. In this case, I'll enable the case in Hugging Face and expose it to our PyTorch team or op team for reproduce. And after their fix, the failures will disappear by only upgrading PyTorch, so it's OK.

Let me know if this philosophy make sense to you, thx.

File: tests/models/nemotron/test_modeling_nemotron.py (hunk 1 above)
Comment (ydshieh) [reply tests/models/nemotron/test_modeling_nemotron.py:216]: Yes, you are right

File: tests/models/bamba/test_modeling_bamba.py [hunk 2]
```diff
@@ -536,22 +547,30 @@ def test_simple_generate(self):
...
-                "!!!<|begin_of_text|>I am late! I need to get to work! I have to get to the",
-            ],
-            9: [
-                "<|begin_of_text|>Hey how are you doing on this lovely evening? I hope you are doing well. I am here",
-                "!!!<|begin_of_text|>I am late! I need to be at the airport in 20 minutes! I",
-            ],
-        }
+        EXPECTED_TEXTS = Expectations(
+            {
+                ("cuda", 7): [],
```
Comment (ydshieh) [inline tests/models/bamba/test_modeling_bamba.py:558]: ok, this is somehow strange, as it means on T4 it will fail, but what happens is GPU OOM instead of out of index . Anyway, I will merge

**Comment by ydshieh** [comment]
I think what @Rocketknight1 originally intend to mean is that we don't want to add tests that we know are failing on our CI. But what @yao-matrix does is to enable tests for XPU, and the mentioned `2 FAILED` are failure on XPU.

Their work won't introduce new failure on our CI runs as we only runs on GPU (Nvidia / AMD).

**Review by ydshieh — APPROVED** [review]
Thank you
